### DISCUSSION TITLE
Self-hosting simplify dependencies

### DISCUSSION THREAD

**Discussion by devMls** [Budibase/budibase#9341] [Ideas]
Hi

I have been reading the self-hosting documentation.

I find that Budibase rely in others products for work, providing those components of funtionality like storage (minIO), database (CouchDB)....

I know that is normal and necessary, but maybe:

-  is possible to abstract this dependecies for their standard interfaces, allowing DevOps teams select his tools?

I try to explain myself

-  maybe change the MinIO dependency for a S3 provider dependency? using the standard S3 API in BudiBase, for alloy the DevOps Team choose what kind of S3 server implement or reuse in his on-premise deployment.

Same for CouchDB or Redis.

What is your opinion?

**Comment by shogunpurple** [comment]
@devMls this is already possible through environment variables - you can set your own S3 access keys, region, redis URL, couch URL etc. It's all completely configurable in self host. See the environment variables in the docker-compose config or `.env` file and tweak them as you like.

https://github.com/Budibase/budibase/blob/master/hosting/docker-compose.yaml

These variables are also supported in other hosting methods such as the single image or kubernetes setups.

**Reply by devMls** [reply]
Thanks for the response @shogunpurple 

Maybe my poor english avoid me explain correctly.

I was trying to propose that the configuration that you mention, allow, for example, introduce for S3 a URL, user and password, of amazon s3 service, that has a S3 API but is not MinIO. 

Or for Redis a URL of Consul, that has a similar API.

Is it supported by budibase?
